### PR TITLE
Make macos_chrome_dev_mode run host-only without device

### PR DESCRIPTION
**Pull request by jmagman**
Remove tethered iOS requirement to allow this test to run on any idle Mac.

Filed https://github.com/flutter/flutter/issues/104034 to also allow `Mac_arm64_ios macos_chrome_dev_mode` to become host-only as well.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jmagman** [comment]
@keyonghan my understanding is that name change will no longer impact benchmarks since they are being uploaded as `macos_chrome_dev_mode`?
https://flutter-flutter-perf.skia.org/e/?queries=test%3Dmacos_chrome_dev_mode

**Comment by jmagman** [comment]
`Windows_android windows_chrome_dev_mode` and `Linux_android linux_chrome_dev_mode` could also be made hostonly.

File: .ci.yaml [hunk 1]
```diff
@@ -3324,13 +3324,14 @@ targets:
...
-  - name: Mac_ios macos_chrome_dev_mode
+  - name: Mac macos_chrome_dev_mode
+    bringup: true
     recipe: devicelab/devicelab_drone
     presubmit: false
     timeout: 60
     properties:
       tags: >
-        ["devicelab", "ios", "mac"]
+        ["devicelab", "hostonly"]
```
Comment (keyonghan) [inline .ci.yaml:3334]: Could we keep the `mac` tag?

**Comment by keyonghan** [comment]
> @keyonghan my understanding is that name change will no longer impact benchmarks since they are being uploaded as `macos_chrome_dev_mode`? https://flutter-flutter-perf.skia.org/e/?queries=test%3Dmacos_chrome_dev_mode

That's correct. If we just filter by `taskName`, then all data will show up. But essentially, data will still be separated as another set due to `device: ios` => `device: none`.

**Comment by jmagman** [comment]
> But essentially, data will still be separated as another set due to device: ios => device: none.

How will that manifest in the benchmark chart? https://flutter-flutter-perf.skia.org/e/?queries=test%3Dmacos_chrome_dev_mode

cc @zanderso

File: .ci.yaml (hunk 1 above)
Comment (jmagman) [reply .ci.yaml:3334]: What does that tag do? It's not on any of the other hostonly tests.
https://github.com/flutter/flutter/blob/5ce07af9ca3cfad2301a4e229b0d30522c82e3ea/.ci.yaml#L3490-L3500

**Comment by jmagman** [comment]
> Remove tethered iOS requirement to allow this test to run on any idle Mac.

Actually @keyonghan, is this true? Or will it only run on Macs without tethered phones?  My intention is to have a larger pool for more tests to run on, but just switching from one pool (with a tethered phone) to another pool (with no tethered phone) doesn't really help, unless the latter pool is much larger.

**Comment by zanderso** [comment]
It sounds like the data will be separated into two separate series when the tags are changed? If this change is isolated to this one benchmark, and we won't be changing it again for a long time, then that is tolerable. I have a feature request open on the Skia side to make this situation easier to navigate: https://bugs.chromium.org/p/skia/issues/detail?id=13161.

File: .ci.yaml (hunk 1 above)
Comment (keyonghan) [reply .ci.yaml:3334]: Let's skip it in this PR then. Created https://github.com/flutter/flutter/issues/104104 to add tag for all host only targets.

Comment (zanderso) [reply .ci.yaml:3334]: Does modifying the tags cause churn on the SkiaPerf side. If so, if we could modify the tags only once, that would be preferable.

Comment (keyonghan) [reply .ci.yaml:3334]: No. Tags here will not affect skia perf. This is what controls the skia perf tags: https://flutter.googlesource.com/recipes/+/refs/heads/main/recipe_modules/test_utils/api.py#171

**Comment by jmagman** [comment]
> Current 90th% queue time for mac/ios is 27min whereas it is ~26min for Mac host only (based on recent cycle). So not much difference. I expect the queue time of host-only bots will drop further, but we need to keep monitoring.

I was going to move all of our `Mac_ios` tests that don't actually use a tethered device to just `Mac` hostonly since they could run on any Mac, but it doesn't sound worth it given current capacity...

**Comment by jmagman** [comment]
Closing, I don't think this is the win I assumed it would be.

**Pull request closed without merging**